src/main/java/com/powsybl/network/conversion/server/NetworkConversionObserver.java

```diff
@@ -0,0 +1,111 @@
+
+    private final MeterRegistry meterRegistry;
+
+    public NetworkConversionObserver(@NonNull ObservationRegistry observationRegistry, @NonNull MeterRegistry meterRegistry) {
+        this.observationRegistry = observationRegistry;
+        this.meterRegistry = meterRegistry;
+    }
+
+    public <T, E extends Throwable> T observeExport(String name, String format, Observation.CheckedCallable<T, E> callable) throws E {
+        // Create an Observation object with the given name and format.
+        Observation observation = createObservation(name, format);
+
+        try {
+            // Observe the execution of the callable within the context of the observation.
+            // The observeChecked method captures the callable's execution details.
+            return observation.observeChecked(() -> {
+                // Call the callable and get the result.
+                T processResult = callable.call();
+
+                // Check if the result is an instance of ExportNetworkInfos.
+                // If it is, calculate the number of buses.
+                if (processResult instanceof ExportNetworkInfos) {
+                    // Add the number of buses as a key-value pair to the observation.
+                    observation.lowCardinalityKeyValue(NUMBER_BUSES, String.valueOf(((ExportNetworkInfos) processResult).getNumberBuses()));
+                }
+                return processResult;
+            });
+
+        } catch (RuntimeException re) {
+            incrementCount(CONVERSION_FAILED_NAME, format);
+            throw re;
+        } catch (Throwable e) {
+            incrementCount(CONVERSION_FAILED_NAME, format);
+            throw (E) e;
+        }
+    }
+
+    public <T, E extends Throwable> T observeImport(String name, String format, Observation.CheckedCallable<T, E> callable) throws E {
+        // Create an Observation object with the given name and format.
+        Observation observation = createObservation(name, format);
+
+        try {
+            // Observe the execution of the callable within the context of the observation.
+            // The observeChecked method captures the callable's execution details.
+            return observation.observeChecked(() -> {
+                // Call the callable and get the result.
+                T processResult = callable.call();
+
+                // Check if the result is an instance of Network.
+                // If it is, calculate the number of buses.
+                if (processResult instanceof Network) {
+                    long numberBuses = ((Network) processResult).getBusView().getBusStream().count();
+                    // Add the number of buses as a key-value pair to the observation.
+                    observation.lowCardinalityKeyValue(NUMBER_BUSES, String.valueOf(numberBuses));
+                }
+                return processResult;
+            });
+
+        } catch (RuntimeException re) {
```

Comment:
Remove all `catch` it already handled by the observation

Reply:
How do you monitor errors with a customized name?

Reply:
See loadfow server

Reply:
In loadflow, we check if result is null, but here if the conversion crashes, we receive an exception